src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -217,6 +231,87 @@ protected static StandardConfiguration getStandardConfigurationFrom(final String
+                .createOptionWithAllEnabled(boundaries);
+
+        // Apply all configurations
+        final String edgeConfiguration = properties.get(EDGE_CONFIGURATION.getName());
+        if (edgeConfiguration != null)
+        {
+            atlasLoadingOption
+                    .setEdgeFilter(getTaggableFilterFrom(edgeConfiguration, sparkContext));
+        }
+
+        final String waySectioningConfiguration = properties
+                .get(WAY_SECTIONING_CONFIGURATION.getName());
+        if (waySectioningConfiguration != null)
+        {
+            atlasLoadingOption.setWaySectionFilter(
+                    getTaggableFilterFrom(waySectioningConfiguration, sparkContext));
+        }
+
+        final String pbfNodeConfiguration = properties.get(PBF_NODE_CONFIGURATION.getName());
+        if (pbfNodeConfiguration != null)
+        {
+            atlasLoadingOption
+                    .setOsmPbfNodeFilter(getTaggableFilterFrom(pbfNodeConfiguration, sparkContext));
+        }
+
+        final String pbfWayConfiguration = properties.get(PBF_WAY_CONFIGURATION.getName());
+        if (pbfWayConfiguration != null)
+        {
+            atlasLoadingOption
+                    .setOsmPbfWayFilter(getTaggableFilterFrom(pbfWayConfiguration, sparkContext));
+        }
+
+        final String pbfRelationConfiguration = properties
+                .get(PBF_RELATION_CONFIGURATION.getName());
+        if (pbfRelationConfiguration != null)
+        {
+            atlasLoadingOption.setOsmPbfRelationFilter(
+                    getTaggableFilterFrom(pbfRelationConfiguration, sparkContext));
+        }
+
+        return atlasLoadingOption;
+    }
+
+    private static Map<String, String> extractAtlasLoadingProperties(final CommandMap command)
+    {
+        final Map<String, String> propertyMap = Maps.newHashMap();
+        propertyMap.put(CODE_VERSION.getName(), (String) command.get(CODE_VERSION));
+        propertyMap.put(DATA_VERSION.getName(), (String) command.get(DATA_VERSION));
+        propertyMap.put(EDGE_CONFIGURATION.getName(), (String) command.get(EDGE_CONFIGURATION));
+        propertyMap.put(WAY_SECTIONING_CONFIGURATION.getName(),
+                (String) command.get(WAY_SECTIONING_CONFIGURATION));
+        propertyMap.put(PBF_NODE_CONFIGURATION.getName(),
+                (String) command.get(PBF_NODE_CONFIGURATION));
+        propertyMap.put(PBF_WAY_CONFIGURATION.getName(),
+                (String) command.get(PBF_WAY_CONFIGURATION));
+        propertyMap.put(PBF_RELATION_CONFIGURATION.getName(),
+                (String) command.get(PBF_RELATION_CONFIGURATION));
+        propertyMap.put(CODE_VERSION.getName(), (String) command.get(CODE_VERSION));
+        propertyMap.put(DATA_VERSION.getName(), (String) command.get(DATA_VERSION));
+
+        return propertyMap;
+    }
+
+    private static Set<Shard> getAllShardsForCountry(final CountryBoundaryMap boundaries,
```

Comment:
Isn't that going to be slow? It would probably benefit from using the grid index, like there:
https://github.com/osmlab/atlas-generator/blob/4.0.4/src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java#L138-L204

Reply:
Yes - that's a good point, let me update to use this!